### ISSUE TITLE
How best to describe the side units?

### ISSUE THREAD

**Issue by terracoda** [phetsims/quadrilateral#296]
One low-vision participant felt that quarters would be easier to understand that "just under" and "just over". Here's what I am working on. Do I increase the number of regions or just the confusing regions with rough quarters. [Draft Table in Design Doc](https://docs.google.com/document/d/1jXayebAWnnNzsT3l6o72YPw4-YtiQaHQNuAi64eiguc/edit#heading=h.u5hfptb25y17).

![image](https://user-images.githubusercontent.com/1902116/208963446-b8bbf31e-3990-48a4-ad6e-fdb982d004e3.png)

**Comment by terracoda** [comment]
@BLFiedler, what do you think of column 3? That's quarters without a change tot he number of regions.

**Comment by terracoda** [comment]
Here's the approximate and exact quarters using "about" as the modifier when the line is on a slant. This removes the "just under" and "just over" modifiers which seemed to be the confusing bits. A finer described grid should provide exact units when the side is parallel or perpendicular to the grid and the "about" units when the side in on a slant. I think that might be nice.

![image](https://user-images.githubusercontent.com/1902116/208991868-c41c65cb-44bb-4afa-a83a-c28b45eb6eb7.png)


![image](https://user-images.githubusercontent.com/1902116/208990474-b1697268-42b4-42ce-aa8c-2f2b8db6e644.png)

**Comment by terracoda** [comment]
I kind of like that the side-lengths would different based on the slant of the line which could be a nice difference if learners notice it.

**Comment by terracoda** [comment]
@jessegreenberg do columns 4 and 5 pose any hurdles?

**Comment by jessegreenberg** [comment]
I don't think so. Just so I understand, describing quarters gets rid of "just under...", "just over...", and "less than..."?

**Comment by terracoda** [comment]
Yes. The table above actually maps out 2 solutions.

In column 3, I have not increased the number regions, I have just changed the words. In column 3 there are no exact quarter regions. There are only "about" quarter regions. There are exact half-regions and hole-regions.

In columns 4 and 5, I updated the words and increased the number regions to include exact quarter regions.

I am wondering if columns 4 and 5 pose any hurdles for implementation? 

I am keen to help implement the strings if that is helpful.

**Comment by jessegreenberg** [comment]
I see! OK, shouldn't be too difficult to implement, but a bit more involved than changing the strings. It will be a new implementation of this function:
https://github.com/phetsims/quadrilateral/blob/35c21bbd4beb9289d274dfc5791477ab4d2eb725/js/quadrilateral/view/SideDescriber.ts#L126-L183

**Comment by terracoda** [comment]
Yes, please! I think this is going to work nicely. Please proceed with columns 4 and 5.

Please note that I wasn't sure about the initial region. I doesn't look like I can make a straight or slanted side under half a unit in length. The description pattern assumes we can get to "about" quarter values on a slant, but I am not that's possible when the side is at min-length (straight or slanted).

If that's the case, the first min-length region might need to be, "about {{one-half}} units" with a range of < or = 1/2 units.

**Comment by BLFiedler** [comment]
Right now I believe this is in what I would call a "round-up" design based on https://github.com/phetsims/quadrilateral/issues/296#issuecomment-1362022288. Anything over the exact scale "rounds up" to be "about" the next unit. Did you want that effect or did you intend there to be even distribution around each quarter scale unit? e.g., "about 1 unit" triggers for >= 7/8 but < 1 or > 1 but less than 1 1/8

Not certain which is better myself, but just wanted to clarify.

**Comment by jessegreenberg** [comment]
@BLFiedler  that is how I interpreted it too. Commit with that assumption made above, back to @terracoda to test.

**Comment by terracoda** [comment]
Yes, I was thinking rounding-up would be good.

**Comment by terracoda** [comment]
I think this sounds really nice! Whenever I check the length of a slanted line, I hear "about" + side units, and whenever I check the length of a straight line aligned to the grid, I hear exact side units.

Honestly, I think the side length units sound awesome.

**I found 1 bug when I use mouse clicks to check the side length.** 
If I click on a side to get the length, and the side is slightly off the grid, the side's vertices sometimes jump to the grid line, changing the shape with a click.

Example: I had this quadrilateral
<img width="585" alt="image" src="https://user-images.githubusercontent.com/1902116/209153239-824c3289-297a-4ecb-88f4-79ee528d1978.png">

And I then did a mouse clicks on on Sides BC and CD, and then DA to check the lengths. Sides BC and CD did not move, but then when I clicked on Side DA corners D and A simultaneously jumped onto the grid.
<img width="583" alt="image" src="https://user-images.githubusercontent.com/1902116/209153423-80da8415-43cc-44a8-a5c4-b97143ef106f.png">

The jumping does not happen all the time. I am going back forth with shift key modifier for mouse and keyboard. It might be connected to that.

I'll put this mouse click bug in a new issue for further investigation.
